### PR TITLE
Complete MetadataState event processor logic.

### PR DESCRIPTION
**Pull request by mariacarmina**
Follow-up for [MetadataState PR](https://github.com/oceanprotocol/ocean-node/pull/135)

Changes proposed in this PR:

- for states 2, 3 -> short version of the DDO
- updated tests

### CODE REVIEW

**Review by paulo-ocean — APPROVED** [review]
lgtm

File: tests/integration/indexer.test.ts [hunk 1]
```diff
@@ -177,10 +177,37 @@ describe('Indexer stores a new published DDO', () => {
...
     expect(retrievedDDO.nft.state).to.equal(parseInt(result[2].toString()))
   })
+
+  it('should change metadata state back to ACTIVE state', async () => {
+    const setMetaDataStateTx = await nftContract.setMetaDataState(0)
+    const trxReceipt = await setMetaDataStateTx.wait()
+    assert(trxReceipt, 'set metada state failed')
+  })
+
+  delay(100000)
```
Comment (jamiehewitt15) [inline tests/integration/indexer.test.ts:192]: This is only testing the contract. We should check that the indexer has updated the state to `0`

File: src/components/database/schemas.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export const schemas: Schemas = {
     {
       name: 'ddo_v0.1',
       enable_nested_fields: true,
-      fields: [{ name: '.*', type: 'auto' }]
+      fields: [{ name: '.*', type: 'auto', optional: true }] // optional: tru for turning the full version DDO into short one for states DEPRECATED and REVOKED
```
Comment (jamiehewitt15) [inline src/components/database/schemas.ts:15]:
```suggestion
      fields: [{ name: '.*', type: 'auto', optional: true }] // optional: true for turning the full version DDO into short one for states DEPRECATED and REVOKED
```

Comment (mariacarmina) [reply src/components/database/schemas.ts]: Fixed this typo.

File: tests/integration/indexer.test.ts (hunk 1 above)
Comment (mariacarmina) [reply tests/integration/indexer.test.ts:192]: I have added the test for this.

**Comment by mariacarmina** [comment]
@jamiehewitt15 can you re-review this, please? Thank you!

**Review by jamiehewitt15 — APPROVED** [review]
Looks good to me

**Pull request merged**